---
title: "EwS-specific neogenes in EwS cell lines (Figure 3-6)"
author: "Lotta Smit"
date: "2024-09-30"
output: html_document
---

```{r setup, include=FALSE}
library(dplyr)
library(tibble)
library(GenomicRanges)
library(DESeq2)
library(ggplot2)
library(ggpubr)
library(gridExtra)
library(ggrepel)
library(tidyverse)
```

```{r, set up: selecting all neogenes}
# load in ORF table
pathtodir <- ""
ORFs <- as.data.frame(read.table(pathtodir, header = TRUE, comment.char = "#", sep = ","))

# create EwS-specific neogene vectors
unannotated <- ORFs %>% 
  filter(gene_biotype == "stringtie", EwS_specific == TRUE, translated == TRUE) %>% 
  pull(gene_id) %>% 
  unique()
lncRNA <- ORFs %>% 
  filter(gene_biotype == "lncRNA", EwS_specific == TRUE, translated == TRUE) %>% 
  pull(gene_id) %>% 
  unique()
EwSs_neogenes <- ORFs %>% filter(gene_biotype %in% c("stringtie", "lncRNA"), EwS_specific == TRUE, translated == TRUE) %>% pull(gene_id) %>% unique()

# create EwS-specific neogene ORF vectors
unannotatedORFs <- ORFs %>% 
  filter(gene_biotype == "stringtie", EwS_specific == TRUE, translated == TRUE) %>% 
  pull(orf_id) %>%
  unique()
lncRNAORFs <- ORFs %>% 
  filter(gene_biotype == "lncRNA", EwS_specific == TRUE, translated == TRUE) %>% 
  pull(orf_id) %>% 
  unique()
EwSs_neoORFs <- ORFs %>% 
  filter(gene_biotype %in% c("stringtie", "lncRNA"), EwS_specific == TRUE, translated == TRUE) %>% 
  pull(orf_id) %>% 
  unique()
```

```{r, set up: identifying neogenes in cell lines}
# identify transcribed EwS-specific neogenes per cell line
  # use res_RNA_cellline from "RNA-seq_EWS_CL.Rmd" 
A673_EwSs_RNA <- res_RNA_A673 %>% 
  filter(gene_id %in% EwSs_neogenes, !is.na(padj)) %>%
  pull(gene_id)
MHHES1_EwSs_RNA <- res_RNA_MHH %>% 
  filter(gene_id %in% EwSs_neogenes, !is.na(padj)) %>%
   pull(gene_id)
SKNMC_EwSs_RNA <- res_RNA_SKNMC %>% 
  filter(gene_id %in% EwSs_neogenes, !is.na(padj)) %>%
   pull(gene_id)
TC106_EwSs_RNA <- res_RNA_TC %>% 
  filter(gene_id %in% EwSs_neogenes, !is.na(padj)) %>%
   pull(gene_id)
RDES_EwSs_RNA <- res_RNA_RDES %>% 
  filter(gene_id %in% EwSs_neogenes, !is.na(padj)) %>%
  pull(gene_id)

# identify total transcribed EwS-specific neogenes
total_transcribed <- unique(c(A673_EwSs_RNA, SKNMC_EwSs_RNA, MHHES1_EwSs_RNA, TC106_EwSs_RNA, RDES_EwSs_RNA))

# get DESeq results of transcribed EwS-specific neogenes of all cell lines
res_EwSs_RNA <- rbind(A673_EwSs_RNA, SKNMC_EwSs_RNA, MHHES1_EwSs_RNA, TC106_EwSs_RNA, RDES_EwSs_RNA) # create cellline_EwSs_RNA without pull(gene_id) first

# identify translated EwS-specific neogenes per cell line
  # Use res_ribo_cellline from "Ribo-seq_EWS_CL.Rmd" 
A673_EwSs_ribo <- res_ribo_A %>% 
  filter(gene_id %in% A673_EwSs_RNA, !is.na(padj)) %>%
  pull(gene_id)
SKNMC_EwSs_ribo <- res_ribo_SKNMC %>% 
  filter(gene_id %in% SKNMC_EwSs_RNA, !is.na(padj)) %>%
   pull(gene_id)
MHHES1_EwSs_ribo <- res_ribo_MHH %>% 
  filter(gene_id %in% MHHES1_EwSs_RNA, !is.na(padj)) %>%
   pull(gene_id)
TC106_EwSs_ribo <- res_ribo_TC %>% 
  filter(gene_id %in% TC106_EwSs_RNA, !is.na(padj)) %>%
   pull(gene_id)
RDES_EwSs_ribo <- res_ribo_RD %>% 
  filter(gene_id %in% RDES_EwSs_RNA, !is.na(padj)) %>%
  pull(gene_id)

# identify total translated EwS-specific neogenes
total_translated <- unique(c(A673_EwSs_ribo, SKNMC_EwSs_ribo, MHHES1_EwSs_ribo, TC106_EwSs_ribo, RDES_EwSs_ribo))

# get DESeq results of translated EwS-specific neogenes of all cell lines
res_EwS_ribo <- rbind(A673_EwSs_ribo, SKNMC_EwSs_ribo, MHHES1_EwSs_ribo, TC106_EwSs_ribo, RDES_EwSs_ribo) # create cellline_EwSs_ribo without pull(gene_id) first
```

```{r Figure 3A}
# Heatmap of transformed normalized counts of all EwS-specific neogenes
  # use dds and res_RNA from "RNA-seq_EWS_CL.Rmd"
# normalize counts
normalized_counts <- counts(dds, normalized = TRUE)

# transform counts into Z-scores
hm <- t(scale(t(normalized_counts)))

# reorder the results obtained earlier
res <- res_RNA[order(res_RNA$pvalue),]
hm <- hm[which(rownames(hm) %in% EwSs_neogenes),]
top_genes <- filter(res, res$gene_id %in% EwSs_neogenes)

# create colors
conditions_cols <- c("#0d31b7", "#E69F00")
class_cols <- c("#009E73", "#E69D70" )
cell_line_cols <- c("#c8e140", "#34b36c",  "#5e63bf", "#ae61a5", "#900C3F")

# create annotations
class_df <- data.frame(geneID = rownames(top_genes), 
                       class = top_genes$class)
annot_row <- data.frame(class = class_df$class[match(rownames(top_genes), class_df$geneID)])
rownames(annot_row) <- top_genes$gene_id

# Force the order of levels we want
annot_row$class <-
  factor(annot_row$class,
         levels = c("lncRNA", "unannotated"))

# Connect the values of the row annotation to the colors
row_cols <-
  list(class = setNames(class_cols,
                        levels(annot_row$class)))
# create row annotation
ha_row <-
  ComplexHeatmap::HeatmapAnnotation(
    df = annot_row,
    show_annotation_name = FALSE,
    which = "row",
    col = row_cols,
    simple_anno_size = unit(2, "mm"),
    annotation_name_gp = gpar(fontsize = 12, fontface = "bold"),
    annot = anno_text(annot_row$class, rot = 45, just = "right", gp = gpar(fontsize = 10)))

# create column annotation
annot_col <- coldata_all_sh
annot_col$conditions <- factor(annot_col$conditions)
annot_col$cell_line <- factor(annot_col$cell_line)
anno_cols <- list(cell_line = setNames(cell_line_cols,
                               levels(annot_col$cell_line)),
                  conditions = setNames(conditions_cols,
                                       levels(annot_col$conditions)))
ha <- HeatmapAnnotation(df = annot_col, 
                        which = "column", 
                        col = anno_cols, 
                        show_annotation_name = FALSE, 
                        simple_anno_size = unit(0.3, "cm"), 
                        show_legend = c(cell_line = FALSE))
# create colors
my_col <- viridis::magma(100)
breaks <- seq(-1.2,2 , length.out = 100)

# create heatmap
heatmap <- ComplexHeatmap::Heatmap(
  hm,
  show_row_dend = FALSE,
  show_column_dend = FALSE,
  cluster_columns = TRUE,
  cluster_rows = TRUE,
  top_annotation = ha,
  column_title_gp = gpar(fontsize = 12, fontface = "bold"),
  column_split = annot_col$cell_line,  
  row_split = annot_row$class,  
  col = circlize::colorRamp2(breaks, my_col),
  heatmap_legend_param = list(
    legend_direction = "horizontal",
    title_position = "topcenter",
    legend_width = unit(10, "line"),
    title = "Gene Z-scores"
  ),
  column_gap = unit(1, "mm"),
  show_column_names = FALSE,
  show_row_names = FALSE)

heatmap <- ComplexHeatmap::draw(
  heatmap,
  heatmap_legend_side = "bottom",
  legend_grouping = "original",
  background = "transparent")

# save heatmap
save(heatmap, file = "heatmap_EwSs_neogenes.png")
```

```{r Figure 3B}
# Volcanoplot of all transcribed EwS-specific neogenes per cell line based on res_EwSs_RNA

# create background lines
hline_data <- data.frame(
  cell_line = c(rep("A673",3), rep( "SK-N-MC", 3), rep("MHH-ES1",2), rep("TC106", 3), rep("RD-ES", 2)),  # Specify the cell_lines you want to apply hlines to
  yintercept = c(20,40,60, 25, 50, 75, 100, 200, 20,40,60, 100, 200))

# calculate counts per cell_line, including total and downregulated genes 
counts <- res_EwSs_RNA %>%
  group_by(cell_line) %>%
  summarise(
    total_points = n(),  
    downregulated = sum(padj < 0.05 & log2FoldChange < -0.75)  
  )

# plot ggplot
 Volcano_EwS_RNA <- ggplot(data = res_EwSs_RNA, aes(x = log2FoldChange, y = -log10(padj), color = class)) + 
  geom_rect(aes(xmin = -Inf, xmax = -0.75, ymin = -log10(0.05), ymax = Inf), fill = "#e7f3fc", alpha = 0.2, color = NA) + 
  geom_vline(xintercept = c(-0.75, 0.75), col = "black", alpha = 0.7, linetype = "dashed") +
  geom_hline(yintercept = -log10(0.01), col = "black", alpha = 0.7, linetype = "dashed") +
  facet_wrap(~cell_line, nrow = 2, scales = "free") +
  geom_hline(data = hline_data, aes(yintercept = yintercept), color = "black", size = 0.5, alpha = 0.2, linetype = "dotted") +
  geom_point() + 
   labs(color = "Gene Class ", size = 0.5, x = "log2FoldChange(DOX+ vs DOX-)") + 
  scale_color_manual(values = c(
    "lncRNA" = "#009E73", 
    "unannotated" = "#E69D70"
  )) +
  geom_label(data = counts, 
            aes(x = Inf, y = Inf, label = paste0("Fusion-driven:\n", downregulated, "/", total_points , "       ")), 
            hjust = 1.05, vjust = 1.1, size = 4, color = "black", 
            fontface = "bold",
            fill = "white") +
   theme_linedraw() +
  theme(panel.grid.major = element_blank(),  
        panel.grid.minor = element_blank(),  
        panel.background = element_rect(fill = 'transparent'), 
        plot.background = element_rect(fill = 'transparent', color = NA), 
        legend.background = element_rect(fill = 'transparent'), 
        legend.box.background = element_rect(fill = 'transparent'), 
        legend.justification = c(0.8,0.15),
        legend.margin = margin(2, 2, 2, 2),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        strip.text = element_text(size = 14, face = "bold")
  ) + 
  guides(colour = guide_legend(override.aes = list(size = 5)))

 Volcano_EwS_RNA
# save volcanoplot
ggsave("Volcano_RNA_EwSs.png", Volcano_EwS_RNA, width = 12)
```


```{r Figure 3C}
# Transcriptionally fusion-driven EwS-s neogenes
# create list of Transcriptionally fusion-driven neogenes by using res_RNA_cellline from "RNA-seq_EWS_CL.Rmd"
SKNMC_enriched_RNA <- res_RNA_SKNMC %>% 
  filter(log2FoldChange < -0.75, padj < 0.05, gene_id %in% EwSs_neogenes) %>%
  pull(gene_id)
MHHES1_enriched_RNA <- res_RNA_MHH %>% 
  filter(log2FoldChange < -0.75, padj < 0.05, gene_id %in% EwSs_neogenes) %>%
  pull(gene_id)
TC106_enriched_RNA <- res_RNA_TC %>% 
  filter(log2FoldChange < -0.75, padj < 0.05, gene_id %in% EwSs_neogenes) %>%
  pull(gene_id)
RDES_enriched_RNA <- res_RNA_RDES %>% 
  filter(log2FoldChange < -0.75, padj < 0.05, gene_id %in% EwSs_neogenes) %>%
  pull(gene_id)
A673_enriched_RNA <- res_RNA_A673 %>% 
  filter(log2FoldChange < -0.75, padj < 0.05, gene_id %in% EwSs_neogenes) %>%
  pull(gene_id)

EwS_enriched_RNA <- list("SK-N-MC" = SKNMC_enriched_RNA, "MHH-ES1" = MHHES1_enriched_RNA, "TC106" = TC106_enriched_RNA, "RD-ES" = RDES_enriched_RNA, "A673" = A673_enriched_RNA)

# prepare upsetplot matrix
m1 <- make_comb_mat(EwS_enriched_RNA)
comb_sizes <- comb_size(m1)
ordered_indices <- order(comb_sizes, decreasing = TRUE)
m1_ordered <- m1[, ordered_indices]
color_scheme <- c("5" = "#d6eaf8", "4" = "#DAF7A6", "3" = "#FFC300", "2" = "#FF5733", "1" = "#900C3F")
comb_degrees <- as.character(comb_degree(m1_ordered))
bar_colors <- color_scheme[comb_degrees]

# plot upsetplot
upset <- UpSet(
  m1_ordered, 
  top_annotation = HeatmapAnnotation(
    "Transcriptionally\n fusion-driven neogenes" = anno_barplot(
      comb_sizes[ordered_indices], 
      border = FALSE, 
      gp = gpar(fill = bar_colors), # Apply custom colors to the bars only
      height = unit(7, "cm")),
    annotation_name_side = "left", 
    annotation_name_rot = 90
  ),
  comb_order = order(comb_degree(-m1_ordered)))

# save upset plot
save(upset, file = "Transcriptionally_fd_EwSs_neogenes.png")
```

```{r Figure 4A}
# Bar plot of transcribed and translated EwS-specific neogenes in cell lines

# create df with all transcribed EwSs neogenes
transcribed_genes <- data.frame(
  gene = total_transcribed,
  SKNMC = as.numeric(total_transcribed %in% SKNMC_EwSs_RNA),
  MHHES1 = as.numeric(total_transcribed %in% MHHES1_EwSs_RNA),
  TC106 = as.numeric(total_transcribed %in% TC106_EwSs_RNA),
  RDES = as.numeric(total_transcribed %in% RDES_EwSs_RNA),
  A673 = as.numeric(total_transcribed %in% A673_EwSs_RNA)
)
rownames(transcribed_genes) <- transcribed_genes$gene
transcribed_genes$gene <- NULL
transcribed_genes <- transcribed_genes %>%
  mutate(sum = rowSums(.))

five <- nrow(transcribed_genes[transcribed_genes$sum == 5,])
four <- nrow(transcribed_genes[transcribed_genes$sum == 4,])
three <- nrow(transcribed_genes[transcribed_genes$sum == 3,])
two <- nrow(transcribed_genes[transcribed_genes$sum == 2,])
one <- nrow(transcribed_genes[transcribed_genes$sum == 1,])

# create df with translated genes
translated_genes <- data.frame(
  gene = total_translated,
  SKNMC = as.numeric(total_translated %in% SKNMC_EwSs_ribo),
  MHHES1 = as.numeric(total_translated %in% MHHES1_EwSs_ribo),
  TC106 = as.numeric(total_translated %in% TC106_EwSs_ribo),
  RDES = as.numeric(total_translated %in% RDES_EwSs_ribo),
  A673 = as.numeric(total_translated %in% A673_EwSs_ribo)
)
rownames(translated_genes) <- translated_genes$gene
translated_genes$gene <- NULL
translated_genes <- translated_genes %>%
  mutate(sum = rowSums(.))

fiv <- nrow(translated_genes[translated_genes$sum == 5,])
fo <- nrow(translated_genes[translated_genes$sum == 4,])
thr <- nrow(translated_genes[translated_genes$sum == 3,])
tw <- nrow(translated_genes[translated_genes$sum == 2,])
on <- nrow(translated_genes[translated_genes$sum == 1,])

# create df with transcribed & translated EwSs neogenes
tr_df <- data.frame(transcribed = c(five, four,three, two, one),
                    translated = c(fiv, fo, thr, tw, on),
                    cell_lines = c(5:1))
tr_long <- pivot_longer(tr_df, cols = c(1:2), values_to = "neogenes", names_to = "class")

# plot barplot
trans_neogenes <- ggplot(tr_long, aes(x = factor(cell_lines), y = neogenes, fill = class)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.8)) + 
  labs(x = "Number of cell Lines", y = "Number of neogenes", fill = "") +
  theme_classic()  +
  scale_fill_manual(values = c("transcribed" = "#b5d4da", "translated" = "#539852")) +
  scale_x_discrete(limits = c("5", "4", "3", "2", "1")) +
  theme(axis.title = element_text(size = 20, face = "bold"),
        axis.text = element_text(size = 15),
        legend.text = element_text(size = 15),
        legend.title = element_text(size = 17, face = "bold"),
        legend.position = "top",
        panel.background = element_rect(fill='transparent'), 
        plot.background = element_rect(fill='transparent', color=NA), 
        legend.background = element_rect(fill='transparent'), 
        legend.box.background = element_rect(fill='transparent')) 
trans_neogenes

# save barplot
ggsave("trans_neogenes.png", trans_neogenes, width = 11.5, bg = "transparent")
```

```{r Figure 4B}
# Translated EwS-specific neogenes

# create list of translated EwS-specific neogenes
EwS_translated <- list(SKNMC = SKNMC_EwSs_ribo, MHHES1 = MHHES1_EwSs_ribo, TC106 = TC106_EwSs_ribo, RDES = RDES_EwSs_ribo, A673 = A673_EwSs_ribo)

# create matrix
m1 <- make_comb_mat(EwS_translated)
m1_ordered <- m1[, order(comb_size(m1), decreasing = TRUE)]
bar_colors <- c("5" = "#d6eaf8", "4" = "#DAF7A6", "3" = "#FFC300", "2" = "#FF5733", "1" = "#900C3F")[as.character(comb_degree(m1_ordered))]

# plot upsetplot
upset <- UpSet(
  m1_ordered, 
  top_annotation = HeatmapAnnotation(
    "Translated EwS-specific neogenes" = anno_barplot(
      comb_sizes[ordered_indices], 
      border = FALSE, 
      gp = gpar(fill = bar_colors), 
      height = unit(7, "cm")),
    annotation_name_side = "left", 
    annotation_name_rot = 90
  ),
  comb_order = order(comb_degree(-m1_ordered)))

# save upsetplot
save(upset, file = "translated_EwSs_neogenes.png")
```

```{r Figure 4C}
# Volcanoplot of all translated EwS-s neogenes in each cell line

# create column with transcriptional fusion-drive 
res_EwS_ribo <- res_EwS_ribo %>%
  mutate(enriched = ifelse(cell_line == "SK-N-MC" & gene_id %in% SKNMC_enriched_RNA, "TRUE", 
                           ifelse(cell_line == "MHH-ES1" & gene_id %in% MHHES1_enriched_RNA, "TRUE",
                                  ifelse(cell_line == "TC106" & gene_id %in% TC106_enriched_RNA, "TRUE",
                                         ifelse(cell_line == "RD-ES" & gene_id %in% RDES_enriched_RNA, "TRUE",
                                                ifelse(cell_line == "A673" & gene_id %in% A673_enriched_RNA, "TRUE", "FALSE"))))))

# create df with background lines
hline_data <- data.frame(
  cell_line = c(rep("A673",3), rep( "SK-N-MC", 4), rep("MHH-ES1",3), rep("TC106", 3), rep("RD-ES", 4)),  
  yintercept = c(3,6,9, 20, 40, 60, 80, 20, 40, 60, 5, 10, 15, 20, 40, 60, 80))

# create counts including total translated neogenes and fusion-driven neogenes
counts <- res_EwS_ribo %>%
  group_by(cell_line) %>%
  summarise(
    total_points = n(),  
    downregulated = sum(padj < 0.05 & log2FoldChange < -0.75))

# plot volcanoplot
Volcano_EwS_Ribo <- ggplot(res_merged_ribo_EwSs,aes(x = log2FoldChange, y = -log10(padj), color = enriched)) +
  scale_color_manual(values = c("TRUE" = "#40af57", "FALSE" = "#dc4141")) + 
  geom_rect(aes(xmin = -Inf, xmax = -0.75, ymin = -log10(0.05), ymax = Inf), fill = "#e7f3fc", alpha = 0.2, color = NA) + 
  facet_wrap(~cell_line, nrow = 2, scales = "free") + 
  geom_hline(data = hline_data, aes(yintercept = yintercept), color = "black", size = 0.5, alpha = 0.2, linetype= "dotted") +
  geom_vline(xintercept = c(-0.75, 1), col = "black", alpha = 0.5, linetype = "dashed") +  
  geom_hline(yintercept = -log10(0.05), col = "black", alpha = 0.5, linetype = "dashed") +
  geom_point(size = 1.7) +
  geom_label(data = counts, 
            aes(x = Inf, y = Inf, label = paste0("Fusion-driven: ",downregulated, "/", total_points)), 
            hjust = 1.04, vjust = 1.3, size = 4, color = "black", 
            fontface = "bold") + 
  theme_linedraw() + 
  theme(panel.grid = element_blank(), 
        panel.background = element_rect(fill = 'transparent'),
        plot.background = element_rect(fill = 'transparent', color = NA), 
        legend.background = element_rect(fill = 'transparent'), 
        legend.box.background = element_rect(fill = 'transparent'), 
        legend.position =  c(0.97,0.1), 
        legend.justification = c(1.2,-0.5),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 14, face = "bold"),
        strip.text = element_text(size = 14, face = "bold")) + 
  labs(color = "Transcriptionally\nFusion-driven", x = "Log2FoldChange (DOX+ vs DOX-)") +
  guides(color = guide_legend(override.aes = list(size = 4)))

# save volcano plot
ggsave("Volcano_ribo_EwSs.png", Volcano_EwS_Ribo, width =  10.5, bg = "transparent")

```

```{r Figure 4D}
# Translationally fusion-driven EwS-s neogenes

# create df of all translationally fusion-driven EwS-s neogenes
SKNMC_enriched_ribo <- res_ribo_S_EwSs %>% 
  dplyr::filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)
MHHES1_enriched_ribo <- res_ribo_M_EwSs %>% 
  dplyr::filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)
TC106_enriched_ribo <- res_ribo_T_EwSs %>% 
  dplyr::filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)
RDES_enriched_ribo <- res_ribo_R_EwSs %>% 
  dplyr::filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)
A673_enriched_ribo <- res_ribo_A_EwSs %>% 
  dplyr::filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)
translationally_fd_genes <- unique(c(A673_enriched_ribo, SKNMC_enriched_ribo, MHHES1_enriched_ribo, TC106_enriched_ribo, RDES_enriched_ribo))
tr_fd_genes.df <- data.frame(
  gene = translationally_fd_genes,
  SKNMC = as.numeric(translationally_fd_genes %in% SKNMC_enriched_ribo),
  MHHES1 = as.numeric(translationally_fd_genes %in% MHHES1_enriched_ribo),
  TC106 = as.numeric(translationally_fd_genes %in% TC106_enriched_ribo),
  RDES = as.numeric(translationally_fd_genes %in% RDES_enriched_ribo),
  A673 = as.numeric(translationally_fd_genes %in% A673_enriched_ribo))

# create df of number of translated neogenes per number of cell lines
tr_fd_genes.df <- tr_fd_genes.df %>%
  mutate(class = ifelse(startsWith(gene, "ENSG"), "lncRNA", "unannotated"),
         sum = rowSums(across(2:6))) %>%
  column_to_rownames("gene")

# plot barplot
transl_fd <- ggplot(tr_fd_genes.df, aes(x = factor(sum), fill = class)) +
  geom_histogram(stat = "count") +
  labs(x = "Number of cell lines", y = "Translationally fusion-driven neogenes ", fill = "Class") +
  theme_classic()  +
  scale_x_discrete(limits = c("5", "4", "3", "2", "1")) +
  theme(axis.title = element_text(size = 20, face = "bold"),
        axis.text = element_text(size = 15),
        legend.text = element_text(size = 15),
        legend.title = element_text(size = 17, face = "bold"),
        legend.position = "top",
        panel.background = element_rect(fill='transparent'), 
        plot.background = element_rect(fill='transparent', color=NA), 
        legend.background = element_rect(fill='transparent'), 
        legend.box.background = element_rect(fill='transparent')) +
  scale_fill_manual(values = c(
    "lncRNA" = "#009E73", 
    "unannotated" = "#E69D70"))

# save barplot
ggsave("translationally_fusiondriven.png", transl_fusionenriched, bg = "transparent", width = 9)
```

```{r Figure 4E}
# Translationally fusion-driven EwS-s neogenes

# create list of driven genes 
SKNMC_enriched_ribo <- res_ribo_S_EwSs %>% 
  filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)
MHHES1_enriched_ribo <- res_ribo_M_EwSs %>% 
  filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)
TC106_enriched_ribo <- res_ribo_T_EwSs %>% 
  filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)
RDES_enriched_ribo <- res_ribo_R_EwSs %>% 
  filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)
A673_enriched_ribo <- res_ribo_A_EwSs %>% 
  filter(log2FoldChange < -0.75, padj < 0.05) %>%
  pull(gene_id)

EwS_enriched_ribo <- list("SK-N-MC" = SKNMC_enriched_ribo, "MHH-ES1" = MHHES1_enriched_ribo, "TC106" = TC106_enriched_ribo, "RD-ES" = RDES_enriched_ribo, "A673" = A673_enriched_ribo)

# plot venn diagram
venn <- ggVennDiagram(
  EwS_enriched_ribo, label_alpha = 0,
  set_color = c("#c1ba1d", "#34b36c",  "#5e63bf", "#c235b6", "#900C3F"),
  edge_size = 1.6,
  label = "count",
  category.names = c("SK-N-MC ", "MHH-ES1", "TC106", "RD-ES", "A673"),
  set_size = 5) +
  scale_fill_gradient(low="grey90",high = "#f93e3e") + 
  labs(fill = "Gene count") +
  theme(legend.title = element_text(size = 16, face = "bold"),
        legend.text = element_text(size = 12),
        legend.position = "right") 

# save venndiagram
ggsave("venn_ribo_fusiondriven.png", venn, bg = "transparent", width = 16)
```

```{r Figure 5A}
# Translation in cell lines and patients on ORF level

# create df of translation in cell lines yes or no
A_transl <- rownames(psitesA)[rownames(psitesA) %in% EwSs_neoORFs]
S_transl <- rownames(psitesSKNMC)[rownames(psitesSKNMC) %in% EwSs_neoORFs]
M_transl <- rownames(psitesMHHES)[rownames(psitesMHHES) %in% EwSs_neoORFs]
T_transl <- rownames(psitesTC)[rownames(psitesTC) %in% EwSs_neoORFs]
R_transl <- rownames(psitesTC)[rownames(psitesTC) %in% EwSs_neoORFs]

patient.data <- ORFs %>% 
  filter(orf_id %in% EwSs_neoORFs) %>%
  select(orf_id, gene_id, n_translated) %>%
  mutate("A673" = ifelse(orf_id %in% A673_transl, 1, 0),
         "SK-N-MC" = ifelse(orf_id %in% S_transl, 1, 0),
         "MHH-ES1" = ifelse(orf_id %in% M_transl, 1, 0),
         "TC106" = ifelse(orf_id %in% T_transl, 1, 0),
         "RD-ES" = ifelse(orf_id %in% R_transl, 1, 0))
patient.data <- patient.data %>%
  mutate(sum = as.factor(rowSums(select(.,"A673", "SK-N-MC", "MHH-ES1", "TC106", "RD-ES"))))

# create count df including total amount of ORFs per number of cell lines
patient.data_summary <- patient.data %>%
  group_by(sum) %>%
  summarise(count = n()) 

# plot violinplot
 violin <- ggplot(patient.data, aes(x = sum, y = n_translated)) + 
   geom_violin(aes(fill = sum)) +
   geom_boxplot(width = 0.1) +
   geom_smooth(aes(x = as.numeric(sum)), method = "lm", se = FALSE, color = "black", size = 0.5, ) +
   geom_jitter(width = 0.1, alpha = 0.5) +
  theme_minimal() +
  scale_fill_manual(values = c("0" = "#930677", "1" = "#900C3F", "2" = "#FF5733", "3" = "#FFC300", "4" = "#DAF7A6", "5" = "#d6eaf8")) +
  geom_text(data = patient.data_summary, 
            aes(x = as.numeric(sum), y = 50, label = paste0("n=", count)), 
            vjust = 0.5, size = 5, color = "black") + 
  labs(x = "Number of cell lines with translation", 
       y = "Number of patients with translation") +
  theme(
    axis.text = element_text(size = 14),
    axis.title = element_text(size = 16, face = "bold"),
    panel.background = element_rect(fill = 'transparent'), 
    plot.background = element_rect(fill = 'transparent', color = NA), 
    legend.background = element_rect(fill = 'transparent'), 
    legend.box.background = element_rect(fill = 'transparent'),
    legend.position = "none") 
 violin
 
 # save violin plot
 ggsave("Violinplot_translation.png", violin, bg ="transparent", height = 8)
```

```{r Figure 5B}
# Selection method of top interesting EwS-s neogenes

# translationally driven in at least 2 cell lines
translationally_driven <- tr_fd_genes.df %>% 
  filter(sum > 1) %>%
  rownames()

# transcriptionally driven in at least 2 cell lines
transcriptionally_driven <- unique(c(A673_enriched_RNA, SKNMC_enriched_RNA, MHHES1_enriched_RNA, TC106_enriched_RNA, RDES_enriched_RNA))
transc_fd.df <- data.frame(
  gene = transcriptionally_driven,
  SKNMC = as.numeric(transcriptionally_driven %in% SKNMC_enriched_RNA),
  MHHES1 = as.numeric(transcriptionally_driven %in% MHHES1_enriched_RNA),
  TC106 = as.numeric(transcriptionally_driven %in% TC106_enriched_RNA),
  RDES = as.numeric(transcriptionally_driven %in% RDES_enriched_RNA),
  A673 = as.numeric(transcriptionally_driven %in% A673_enriched_RNA))
transcriptionally_driven <- transc_fd.df %>%
  mutate(sum = rowSums(select(.,2:6))) %>%
  filter(sum > 1) %>%
  pull(gene)

# combined gives final list of most interesting EwS-s neogenes
t_t_driven <- translationally_driven[translationally_driven %in% transcriptionally_driven] 

# translated (translated_genes from Figure 4A)
translated_genes <- translated_genes %>%
  filter(sum > 1) %>%
  rownames()

# plot venn diagram
selection.method <- ggVennDiagram(list(transcriptionally=transcriptionally_driven, translationally = translationally_driven, translated = translated_genes), label_alpha = 0,
  set_color = c("#406ecb","#075f09",  "#599c68"),
  edge_size = 1.6,
  label = "count",
  category.names = c("Transcriptionally\nfusion-driven (n=44)", "Translationally\nfusion-driven (n=22)", "Translated (n=47)"),
  set_size = 2.6) +
  scale_fill_gradient(low="grey90",high = "#ed3d3d") + 
  labs(fill = "Gene count") +
  theme(legend.title = element_text(size = 16, face = "bold"),
        legend.text = element_text(size = 12),
        legend.position = "right") 

# save venn diagram
ggsave("Selectionmethod.png", selection.method, height = 10)
```

```{r Figure 6A/B}
# Transcription = normalized counts of top 6 / Translation = normalized p-sites of top 6
  # use the dds from "RNA-seq_EWS_CL.Rmd" & "Ribo-seq_EWS_CL.Rmd" to visualize transcription & translation, respectively
  
# RNA:
dds <- DESeqDataSetFromTximport(txi = txi_all_sh, colData = coldata_all_sh, design = ~ cell_line + conditions)
# Ribo:
dds <- DESeqDataSetFromMatrix(countData = merged_psites_all_sh, colData = coldata_all_sh, design = ~ cell_line+conditions)

# continue with the same script: create df of top 6 counts / p-sites
dds <- DESeq(dds)

XLOC_036825 <- plotCounts(dds, gene = "XLOC_036825", intgroup = c("conditions", "cell_line"), returnData = TRUE)
XLOC_053066  <- plotCounts(dds, gene="XLOC_053066", intgroup=c("conditions", "cell_line"), returnData=TRUE)
XLOC_101091  <- plotCounts(dds, gene="XLOC_101091", intgroup=c("conditions", "cell_line"), returnData=TRUE)
XLOC_107598  <- plotCounts(dds, gene="XLOC_107598", intgroup=c("conditions", "cell_line"), returnData=TRUE)
XLOC_013861 <- plotCounts(dds, gene="XLOC_013861", intgroup=c("conditions", "cell_line"), returnData=TRUE)
ENSG00000251460 <- plotCounts(dds, gene="ENSG00000251460", intgroup=c("conditions", "cell_line"), returnData=TRUE)

XLOC_036825$gene <- "XLOC_036825"
XLOC_053066$gene <- "XLOC_053066"
XLOC_101091$gene <- "XLOC_101091"
XLOC_107598$gene <- "XLOC_107598"
XLOC_013861$gene <- "XLOC_013861" 
ENSG00000251460$gene <- "ENSG00000251460" 

top6.df <- rbind(XLOC_013861,XLOC_036825, XLOC_053066, XLOC_101091, XLOC_107598, ENSG00000251460)


# plot box plots
EwSsCounts <- ggplot(top6.df, aes(x= cell_line, y=count, fill = conditions, color = conditions)) + 
  geom_boxplot(aes(color = conditions)) + 
  theme_linedraw() + 
  facet_wrap(.~gene, scales = "free", nrow = 2) + 
  scale_fill_manual(values = c("#003A70", "#FFA500")) + 
  scale_color_manual(values = c("#004F70", "#DA7E1C")) + 
  labs(color = "Conditions", y= "Normalized counts", x= " ", title = "Transcription") +
  theme(panel.background = element_rect(fill='transparent'),
        plot.background = element_rect(fill='transparent', color=NA), 
        legend.background = element_rect(fill='transparent'), 
        legend.box.background = element_rect(fill='transparent'),
        text = element_text(size = 14),
        axis.title = element_text(size = 16, face = "bold"), 
        strip.text = element_text(size = 15, face = "bold"),
        legend.position =  "bottom",
        axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
        title = element_text(size = 16, face = "bold")) +
  scale_y_sqrt() 

# save boxplot
ggsave("counts_All_EwSsneogenes.png", EwSsCounts, width = 12, height = 10)
```

